### PR TITLE
M4 - Build merkle tree from genesis ledger and verify root hash

### PR DESCRIPTION
**Pull request by hanabi1224**
**Summary of changes**
Changes introduced in this pull request:
- Add fixed-height support to `MinaMerkleLedger` (Mina uses fixed height 20 no matter how many accounts a tree stores, and the merkle root is the hash at height 20)
- Properly calculate root hash when fixed-height is set
- Verify the root hash of the Merkle tree built from the genesis ledger

**Reference issue to close (if applicable)**

This closes the merger part of https://github.com/ChainSafe/mina-rs/issues/130

Closes 

**Other information and links**

### CODE REVIEW

File: merkle/src/tree_impl.rs [hunk 1]
```diff
@@ -119,7 +127,25 @@ where
...
+        let hash = self.calculate_hash_if_needed(0);
+        if let Some(fixed_height) = self.fixed_height {
+            match fixed_height.cmp(&self.height) {
+                Ordering::Less => panic!(
+                    "fixed_height {} should not be smaller than current height {}",
+                    fixed_height, self.height,
+                ),
+                Ordering::Equal => hash,
+                Ordering::Greater => {
+                    let mut hash = hash;
```
Comment (creativcoder) [inline merkle/src/tree_impl.rs:139]: can make hash mut at `130` instead

**Review by creativcoder — APPROVED** [review]
:+1:

File: merkle/src/tree_impl.rs (hunk 1 above)
Comment (hanabi1224) [reply merkle/src/tree_impl.rs]: Fixed

**Review by willemolding — APPROVED** [review]

File: merkle/src/tree_impl.rs [hunk 2]
```diff
@@ -45,6 +46,13 @@ where
         }
     }
 
+    /// Sets the fixed height of the merkle tree
+    /// Note that leaf nodes are at height 0 here but -1 in mina OCaml implementation
+    pub fn with_fixed_height(mut self, height: u32) -> Self {
```
[the remark this answers is not in the document]
Comment (hanabi1224) [reply merkle/src/tree_impl.rs]: That's a great suggestion, thx!

**Pull request merged**